### PR TITLE
New System: Enhanced File Checker

### PR DESCRIPTION
**Pull request by D1GQ**
# Overview
Added a new system called the ```Enhanced File Checker```, this checks the game files of Among Us and checks what BepInEx mods are running in the process, if it detects any unauthorized files or mods it will deny access to online play, and later on it is planned to send information to the API of the of the players Name/FriendCode/PUID and tags of what got detected. This should add an extra layer of protection people trying to cheat while using the mod!
Is_a_cheater
## Futures:
```diff
+ | New and improved EAC List check.
+ | Disable play button until EAC list it's gathered.
+ | BepInEx mod ban list.
+ | Among Us version text word checker / ban list.
+ | Sicko, Among Us Menu, and Malum Menu detection.
+ | Player information logger.
! | Fixed EAC bypass.
! | And much more!
- | This system will automatically be disabled for dev's for testing the anti-cheat.
```

### CODE REVIEW

**Comment by Ultradragon005** [comment]
couldn't they just build the source code without it tho..

**Comment by D1GQ** [comment]
> couldn't they just build the source code without it tho..

The average Joe wouldn't really know how to do that though.

**Comment by Pietrodjaowjao** [comment]
why moe keeps commiting here

**Comment by Ultradragon005** [comment]
Nah, I think it's because merging, makes the commit appear on the branch

**Comment by D1GQ** [comment]
> Nah, I think it's because merging, makes the commits appear on the branch

That is correct

File: Patches/CredentialsPatch.cs [hunk 1]
```diff
@@ -214,6 +214,7 @@ public static void Prefix(ModManager __instance)
 
         LateTask.Update(Time.deltaTime);
         CheckMurderPatch.Update();
+        EFC.UpdateUnauthorizedFiles();
```
Comment (Tommy-XL) [inline Patches/CredentialsPatch.cs:217]: This code will be constantly checked during the game, won't it?

Comment (Tommy-XL) [reply Patches/CredentialsPatch.cs:217]: I mean I don't want it to affect performance when the mod is constantly reproducing this code

Comment (D1GQ) [reply Patches/CredentialsPatch.cs:217]: All the code is doing is checking if it did find an unauthorized file, checking for the unauthorized files only gets run when someone presses on the play button, so I don't think it should decrease the performance, but if it's a big issue I can move it back to the update on the main menu

Comment (Pietrodjaowjao) [reply Patches/CredentialsPatch.cs:217]: theres no reason to do this, mods cant be added in runtime lol

Comment (Pietrodjaowjao) [reply Patches/CredentialsPatch.cs:217]: Please patch MMOnlineManager.Start or something like this

Comment (Tommy-XL) [reply Patches/CredentialsPatch.cs:217]: Yeah, this will probably be better than a constant check that is checked 20-30 times in one second

Comment (D1GQ) [reply Patches/CredentialsPatch.cs:217]: I'll do something to take down the stress

**Comment by 0xDrMoe** [comment]
This can be closed due to BAU now, correct?

**Comment by D1GQ** [comment]
Yeah

**Pull request closed without merging**